src/test/java/uk/gov/companieshouse/api/accounts/transformer/CreditorsAfterOneYearTransformerTest.java

```diff
@@ -0,0 +1,144 @@
+        assertEquals(new HashMap<>(), creditorsAfterOneYearEntity.getData().getLinks());
+    }
+
+    @Test
+    @DisplayName("Tests transformer with empty previous period Rest Object")
+    public void testRestToEntityTransformerWithEmptyPreviousPeriodRestObject() {
+
+        CreditorsAfterOneYear creditorsAfterOneYear = new CreditorsAfterOneYear();
+
+        creditorsAfterOneYear.setEtag(ETAG);
+        creditorsAfterOneYear.setKind(KIND);
+        creditorsAfterOneYear.setLinks(new HashMap<>());
+
+        CurrentPeriod currentPeriod = createCurrentPeriodRestObject();
+
+        creditorsAfterOneYear.setCurrentPeriod(currentPeriod);
+
+        CreditorsAfterOneYearEntity creditorsAfterOneYearEntity = creditorsAfterOneYearTransformer
+                .transform(creditorsAfterOneYear);
+
+        assertNotNull(creditorsAfterOneYearEntity);
+        assertNull(creditorsAfterOneYearEntity.getData().getPreviousPeriodEntity());
+        assertEqualsEntityObject(creditorsAfterOneYearEntity);
+        assertEquals(new HashMap<>(), creditorsAfterOneYearEntity.getData().getLinks());
+    }
+
+    @Test
+    @DisplayName("Tests transformer with fully populated Rest object and validates values returned")
+    public void testRestToEntityTransformerWithFullyPopulatedObject() {
+
+        CreditorsAfterOneYear creditorsAfterOneYear = new CreditorsAfterOneYear();
+
+        creditorsAfterOneYear.setEtag(ETAG);
+        creditorsAfterOneYear.setKind(KIND);
+        creditorsAfterOneYear.setLinks(new HashMap<>());
+
+        creditorsAfterOneYear.setCurrentPeriod(createCurrentPeriodRestObject());
+        creditorsAfterOneYear.setPreviousPeriod(createPreviousPeriodRestObject());
+
+        CreditorsAfterOneYearEntity creditorsAfterOneYearEntity = creditorsAfterOneYearTransformer
+                .transform(creditorsAfterOneYear);
+
+        assertNotNull(creditorsAfterOneYearEntity);
+        assertEqualsEntityObject(creditorsAfterOneYearEntity);
+    }
+
+    private CurrentPeriod createCurrentPeriodRestObject() {
+
+        CurrentPeriod currentPeriod = new CurrentPeriod();
+        currentPeriod.setBankLoansAndOverdrafts(BANK_LOANS_AND_OVERDRAFTS);
+        currentPeriod.setOtherCreditors(OTHER_CREDITORS);
+        currentPeriod.setFinanceLeasesAndHirePurchaseContracts(FINANCE_LEASE);
+        currentPeriod.setTotal(TOTAL);
+        currentPeriod.setDetails(DETAILS);
+
+        return currentPeriod;
+    }
+
+
+    private PreviousPeriod createPreviousPeriodRestObject() {
+
+        PreviousPeriod previousPeriod = new PreviousPeriod();
+        previousPeriod.setBankLoansAndOverdrafts(BANK_LOANS_AND_OVERDRAFTS);
+        previousPeriod.setOtherCreditors(OTHER_CREDITORS);
+        previousPeriod.setFinanceLeasesAndHirePurchaseContracts(FINANCE_LEASE);
+        previousPeriod.setTotal(TOTAL);
+
+        return previousPeriod;
+    }
+
+    private void assertEqualsEntityObject(CreditorsAfterOneYearEntity creditorsAfterOneYearEntity) {
```

Comment:
Can we rename this method to be a little more explicit? Something like `assertFieldsMappedToEntity` or similar?